### PR TITLE
[GitLab] Improvements to MR details.

### PR DESCRIPTION
**Pull request by metakirby5**
## Description

- Add `TagList` items to MR details to show assignees and reviewers.
Note that `Assignee` and `Reviewer` are not pluralized in GitLab, so I have chosen to not pluralize it in the `TagList` as well.
- Hide metadata items when they are empty.
- Add `key` properties where necessary.

## Screencast

Anonymized some bits.

Screen Shot 2022-06-28 at 23 39 49

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/gitlab/src/components/mr.tsx [hunk 1]
```diff
@@ -113,21 +111,33 @@ export function MRDetail(props: { mr: MergeRequest }): JSX.Element {
...
           <Detail.Metadata.Label title="Into" text={mr.target_branch} />
-          <Detail.Metadata.Label title="Author" text={author} />
-          <Detail.Metadata.Label title="Milestone" text={milestone} />
-          <Detail.Metadata.TagList title="Labels">
-            {mr.labels.map((m) => (
-              <Detail.Metadata.TagList.Item text={m.name} color={m.color} />
-            ))}
-          </Detail.Metadata.TagList>
+          {mr.author && <Detail.Metadata.Label title="Author" text={mr.author.name} />}
+          {!!mr.assignees.length && (
```
Comment (Loskir) [inline extensions/gitlab/src/components/mr.tsx:119]:
```suggestion
          {mr.assignees.length > 0 && (
```

File: extensions/gitlab/src/components/mr.tsx [hunk 2]
```diff
@@ -113,21 +111,33 @@ export function MRDetail(props: { mr: MergeRequest }): JSX.Element {
...
+            <Detail.Metadata.TagList title="Assignee">
+              {mr.assignees.map((a) => (
+                <Detail.Metadata.TagList.Item key={a.id} text={a.name} icon={a.avatar_url} />
+              ))}
+            </Detail.Metadata.TagList>
+          )}
+          {!!mr.reviewers.length && (
```
Comment (Loskir) [inline extensions/gitlab/src/components/mr.tsx:126]:
```suggestion
          {mr.reviewers.length > 0 && (
```

File: extensions/gitlab/src/components/mr.tsx [hunk 3]
```diff
@@ -113,21 +111,33 @@ export function MRDetail(props: { mr: MergeRequest }): JSX.Element {
...
+            <Detail.Metadata.TagList title="Reviewer">
+              {mr.reviewers.map((a) => (
+                <Detail.Metadata.TagList.Item key={a.id} text={a.name} icon={a.avatar_url} />
+              ))}
+            </Detail.Metadata.TagList>
+          )}
+          {mr.milestone && <Detail.Metadata.Label title="Milestone" text={mr.milestone.title} />}
+          {!!mr.labels.length && (
```
Comment (Loskir) [inline extensions/gitlab/src/components/mr.tsx:134]:
```suggestion
          {mr.labels.length > 0 && (
```

**Review by Loskir — APPROVED** [review]
Great! Some suggestions

**Comment by tonka3000** [comment]
@metakirby5 Nice addition. Could you also add assignee to the issue detail metadata view?

**Comment by metakirby5** [comment]
@Loskir Thanks!
@tonka3000 Just added assignees, and also author avatar icons. Please take a look!

**Review by tonka3000 — APPROVED** [review]

**Comment by metakirby5** [comment]
FYI: Added some visual changes to make the user displays more pleasing 😄

image

**Review by pernielsentikaer — APPROVED** [review]
@metakirby5  - Thanks for your contribution 🔥

**Pull request merged**